libosmscout-import/src/osmscout/import/GenLocationIndex.cpp

```diff
@@ -94,8 +95,26 @@ namespace osmscout {
     return false;
   }
 
-  bool LocationIndexGenerator::Region::CouldContain(const Region& region) const
+  bool LocationIndexGenerator::Region::CouldContain(const Region& region, bool strict) const
   {
+    if (strict) {
+      // test if the region can be fit into *this when the boundingBox
+      // is taken into account
+      double area_region = region.boundingBox.GetSize(); // reference area
+
+      GeoCoord cornerMin( std::max( region.boundingBox.GetMinLat(), this->boundingBox.GetMinLat() ),
+			  std::max( region.boundingBox.GetMinLon(), this->boundingBox.GetMinLon() ) );
+      
+      GeoCoord cornerMax( std::min( region.boundingBox.GetMaxLat(), this->boundingBox.GetMaxLat() ),
+			  std::min( region.boundingBox.GetMaxLon(), this->boundingBox.GetMaxLon() ) );
+
+      GeoBox bx(cornerMin, cornerMax);
```

Comment:
I suggest to create new method `Intersection` to `GeoBox` for this. I was surprised that it don't exists yet :-)

```
  /**
   * Create new GeoBox from intersection of this with other
   * If not Intersects, invalid GeoBox is returned
   * @param other
   */
  GeoBox GeoBox::Intersection(const GeoBox& other) const {
    assert(valid);
    assert(other.valid);

    if (!Intersects(other)){
      return GeoBox();
    }

    GeoCoord cornerMin( std::max( other.GetMinLat(), GetMinLat()),
                        std::max( other.GetMinLon(), GetMinLon()));

    GeoCoord cornerMax( std::min( other.GetMaxLat(), GetMaxLat()),
                        std::min( other.GetMaxLon(), GetMaxLon()));

    return GeoBox(cornerMin, cornerMax);
  }

```